src/index.ts

```diff
@@ -75,6 +201,70 @@ export default class BlueButton {
         config.environment === Environments.PRODUCTION
           ? PRODUCTION_BASE_URL
           : SANDBOX_BASE_URL,
+      pkce: config.pkce ? config.pkce : true,
+      retryConfig: config.retryConfig,
     };
   }
+
+  async getResource(fhirReq: FhirRequest) {
+    const user = fhirReq.getAppContext().getUser();
+
+    let accessToken = user.getAuthorizationToken();
+
+    if ( user.accessTokenExpired() ) {
+        accessToken = await accessTokenRefresh(fhirReq.getAppContext());
+    }
+
+    let fhirUrl;
+
+    switch (fhirReq.getFhirResourceType()) {
+        case FhirResourceType.Patient:
+            fhirUrl = fhirReq.getAppContext().getBlueButton().BB2_PATIENT_URL;
+            break;
+        case FhirResourceType.Coverage:
+            fhirUrl = fhirReq.getAppContext().getBlueButton().BB2_COVERAGE_URL;
+            break;
+        case FhirResourceType.ExplanationOfBenefit:
+            fhirUrl = fhirReq.getAppContext().getBlueButton().BB2_EOB_URL;
+            break;
+        default:
+            throw Error(`Unknown Fhir Resource Type --> ${fhirReq.getFhirResourceType()}`)
+    }
+
+    const response = await get(fhirUrl, fhirReq.getRequest().query, 
```

Comment:
I'm very confused about using express request/response objects here. Express is a web server framework. So Express request/response represents a users request to your API and the response is what you are giving back. We should be using axios here - not express. Since this function should be fetching data from the BB2 API.

Reply:
will re-factor out the express dependency.